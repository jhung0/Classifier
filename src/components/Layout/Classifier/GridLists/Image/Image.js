import React, { Component } from 'react';
import { DragSource } from 'react-dnd';
import ItemTypes from '../../ItemTypes';

//const handlePictureDrag = this.props.handlePictureDrag


const imSource = {
    beginDrag(props) {
      return {  filename: props.fileName,
                grid: props.grid
             };
    },
    endDrag(props, monitor, component){
        const imInfo = {target: monitor.getDropResult().grid,  origin: props.grid, fileName: props.fileName }
        component.imageHandlePictureDrag(imInfo)
    }
};
  
function collect(connect, monitor) {
    return { connectDragSource: connect.dragSource(),
             isDragging: monitor.isDragging()
    }
  };

class Image extends Component {

    imageHandlePictureDrag (imInfo) {
      this.props.handlePictureDrag(imInfo)
    }
    
    render(){
      const { connectDragSource } = this.props;   
      return  connectDragSource(<img grid={this.props.grid} fileName={this.props.fileName} src={this.props.src} width={48} alt=""/>)       
  }
};

export default   DragSource(ItemTypes.IM, imSource, collect)(Image);