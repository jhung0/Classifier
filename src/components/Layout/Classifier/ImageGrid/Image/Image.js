import React, { Component } from 'react';
import { DragSource } from 'react-dnd';
//import './image.css'
import ItemTypes from '../../../../ItemTypes';


const imSource = {
    beginDrag(props) {
      return {  filename: props.fileName,
                grid: props.grid
             };
    },
    endDrag(props, monitor, component){
        if(monitor.didDrop()){
          const imInfo = {target: monitor.getDropResult().grid,  origin: props.grid, fileName: props.fileName }
          component.imageHandlePictureDrop(imInfo)
        }
    }
};
  
function collect(connect, monitor) {
    return { connectDragSource: connect.dragSource(),
             isDragging: monitor.isDragging()
    }
  };

class Image extends Component {

    state = {
      hovering: false
    }

    imageHandlePictureDrop (imInfo) {
      this.props.handlePictureDrop(imInfo);
    }

    onhover (mythis) {   
      mythis.setState({hovering: true})
    }

    onhoverOut (mythis) {   
      mythis.setState({hovering: false})

    }

    render(){
      
      const { connectDragSource } = this.props;   
      return  connectDragSource(
                                <img 
                                     //onMouseOver={() => this.onhover(this)}
                                     //onMouseOut={() => this.unhover(this)}   
                                     grid={this.props.grid} 
                                     src={this.props.src} 
                                     width={48} alt=""/>
                                )    
  }

};

export default   DragSource(ItemTypes.IM, imSource, collect)(Image);