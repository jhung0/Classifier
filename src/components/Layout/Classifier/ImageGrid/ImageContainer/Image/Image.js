import React, { Component } from "react";
import { DragSource } from "react-dnd";
import ItemTypes from "../../../../../ItemTypes";

const imSource = {
  beginDrag(props) {
    return {
      id: props.id,
      category: props.category
    };
  },

  endDrag(props, monitor, component) {
    if (monitor.didDrop()) {
      const catId = monitor.getDropResult().targetCat;
      const imageId = Number(props.id);
      const dropInfo = {
        catId: catId,
        imageId: imageId
      };
      component.imageHandlePictureDrop(dropInfo);
    }
  }
};

function collect(connect, monitor) {
  return {
    connectDragSource: connect.dragSource(),
    isDragging: monitor.isDragging()
  };
}

class Image extends Component {
  state = {
    hovering: false
  };

  imageHandlePictureDrop(dropInfo) {
    this.props.drop(dropInfo);
  }

  onhover(mythis) {
    mythis.setState({ hovering: true });
  }

  onhoverOut(mythis) {
    mythis.setState({ hovering: false });
  }

  shouldComponentUpdate (nextProps, nextState){
    return false
  }

  render() {
    const { connectDragSource } = this.props;
    return connectDragSource(
      <img
        //onMouseOver={() => this.onhover(this)}
        //onMouseOut={() => this.unhover(this)}
        id={this.props.id}
        src={this.props.src}
        category={this.props.accuracy}
        width={"100%"}
        alt={"Ups"}
        style={{ height: "auto", marginRight: "5px", marginBottom: "35px" }}
      />
    );
  }
}

export default DragSource(ItemTypes.IM, imSource, collect)(Image);
