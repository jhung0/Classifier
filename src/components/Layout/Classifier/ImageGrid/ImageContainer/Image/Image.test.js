import React from 'react';
import Image from './Image';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const OriginalImage = Image.DecoratedComponent;

    const identity = el => el;

    const tree = renderer
        .create(<OriginalImage connectDragSource={identity} />)
        .toJSON();

    expect(tree).toMatchSnapshot();
});
