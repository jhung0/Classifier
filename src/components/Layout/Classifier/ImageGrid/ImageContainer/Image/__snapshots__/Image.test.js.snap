// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<img
  alt="Ups"
  category={undefined}
  id={undefined}
  src={undefined}
  style={
    Object {
      "height": "auto",
      "marginBottom": "35px",
      "marginRight": "5px",
    }
  }
  width="100%"
/>
`;
